pkg/dependencies/factory.go

```diff
@@ -130,17 +130,38 @@ func (f *Factory) WithRegistryMirror(mirror string) *Factory {
 	return f
 }
 
-func (f *Factory) WithExecutableImage(image string) *Factory {
+func (f *Factory) UseExecutableImage(image string) *Factory {
 	f.executablesImage = image
 	return f
 }
 
+func (f *Factory) WithExecutableImage() *Factory {
+	f.WithManifestReader()
+
+	f.buildSteps = append(f.buildSteps, func(ctx context.Context) error {
+		if f.executablesImage != "" {
+			return nil
+		}
+
+		bundles, err := f.dependencies.ManifestReader.ReadBundlesForVersion(version.Get().GitVersion)
+		if err != nil {
+			return fmt.Errorf("retrieving executable tools image from bundle in dependency factory: %v", err)
+		}
+
+		f.executablesImage = bundles.Spec.VersionsBundles[0].Eksa.CliTools.VersionedImage()
```

Comment:
are the cli image all same across k8s versions?

Reply:
Right now they all are. I don't see us building different versions of this image but there are no guarantees and the bundles does support that case.

This is just the "default" for commands than don't take a cluster config file (hence we don't have a kubernetes version)
All the ones that do take it (create, upgrade, delete...) will use the correct one for their kubernetes version

Reply:
Offline discussion: write a comment